iModelCore/ecobjects/test/NonPublished/CustomAttributeConversionsTest.cpp

```diff
@@ -2133,6 +2133,217 @@ TEST_F(StandardValueToEnumConversionTest, TestRootClassInSeparateSchema)
+            <ECClass typeName='TestClass' isDomainClass='True'>
+            </ECClass>
+        </ECSchema>)xml";
+
+    Utf8CP schemaXml = R"xml(<?xml version='1.0' encoding='UTF-8'?>
+        <ECSchema schemaName='test' version='1.0' nameSpacePrefix='ts' xmlns='http://www.bentley.com/schemas/Bentley.ECXML.2.0'>
+            <ECSchemaReference name='EditorCustomAttributes' version='01.00' prefix='beca' />
+            <ECSchemaReference name='Middle' version='01.00' prefix='mid'/>
+            <ECClass typeName='TestClass' isDomainClass='True'>
+                <BaseClass>mid:TestClass</BaseClass>
+                <ECProperty propertyName='testProp' typeName='int'>
+                    <ECCustomAttributes>
+                        <StandardValues xmlns='EditorCustomAttributes.01.00'>
+                            <ValueMap>
+                                <ValueMap>
+                                    <Value>0</Value>
+                                    <DisplayString>value0</DisplayString>
+                                </ValueMap>
+                                <ValueMap>
+                                    <Value>1</Value>
+                                    <DisplayString>value1</DisplayString>
+                                </ValueMap>
+                                <ValueMap>
+                                    <Value>2</Value>
+                                    <DisplayString>value2</DisplayString>
+                                </ValueMap>
+                            </ValueMap>
+                        </StandardValues>
+                    </ECCustomAttributes>
+                </ECProperty>
+            </ECClass>
+        </ECSchema>)xml";
+ 
+    ECSchemaPtr baseSchema;
+    ECSchemaPtr middleSchema;
+    ECSchemaPtr schema;
+    ECSchemaReadContextPtr context = ECSchemaReadContext::CreateContext();
+    ASSERT_EQ(SchemaReadStatus::Success, ECSchema::ReadFromXmlString(baseSchema, baseSchemaXml, *context));
+    ASSERT_TRUE(baseSchema.IsValid());
+    ASSERT_EQ(SchemaReadStatus::Success, ECSchema::ReadFromXmlString(middleSchema, middleSchemaXml, *context));
+    ASSERT_TRUE(middleSchema.IsValid());
+    ASSERT_EQ(SchemaReadStatus::Success, ECSchema::ReadFromXmlString(schema, schemaXml, *context));
+    ASSERT_TRUE(schema.IsValid());
+ 
+    EXPECT_TRUE(ECSchemaConverter::Convert(*schema.get(), *context.get())) << "Schema conversion failed";
+ 
+    ASSERT_EQ(1, schema->GetEnumerationCount()) << "The number of enumerations created is not as expected.";
+    ASSERT_EQ(0, middleSchema->GetEnumerationCount()) << "The number of enumerations created is not as expected.";
+    ASSERT_EQ(0, baseSchema->GetEnumerationCount()) << "The number of enumerations created is not as expected.";
+
+    EXPECT_EQ(nullptr, middleSchema->GetClassP("TestClass")->GetPropertyP("testProp"));
+    CheckTypeName("TestClass_testProp", *schema, "testProp", {"TestClass"});
+    CheckTypeName("int", *baseSchema, "testProp", {"TestClass"});

```

Comment:
This is wrong and I think highlights that the solution proposed is fundamentally wrong.  All properties in the hierarchy should be converted to use the enumeration and the enumeration must be defined in the base schema.  

If a property override exists that uses this enum then the schema must have a direct reference to the schema defining the enum

Reply:
I initially planned to remove only the property from the `mid:TestClass` class, but I accidentally removed the base class as well. I have updated the test case, and we can see that the enum is only created in the base schema.